gems/pending/openstack/openstack_handle/handled_list.rb

```diff
@@ -0,0 +1,62 @@
+require 'openstack/openstack_handle/pagination/marker'
+require 'openstack/openstack_handle/pagination/page_number'
+require 'openstack/openstack_handle/pagination/none'
+
+require 'openstack/openstack_handle/multi_tenancy/loop'
+
+module OpenstackHandle
+  module HandledList
+    def handled_list(collection_type, options = {})
+      # Will automatically handle multi-tenancy and pagination of all Fog list methods, so we always get all openstack
+      # entities back. The exceptions of each service and collection type will be solved in <service_name>)_delegate
+      # classes by defining multi_tenancy_type and pagination_type methods base on collection_type
+      # Example of call in refresh code
+      # @compute_service.handled_list(:servers)
+      # @orchestration_service.handled_list(:resources, :stack => stack)
+      # By default, it calls :all method on the Fog collection, that has unified interface in all list methods in Fog
+      # and always returns detailed list.
+      multi_tenancy_class.new(self, @os_handle, self.class::SERVICE_NAME, collection_type, options,
+                              :all).list
+    rescue Excon::Errors::Forbidden => err
+      # It can happen user doesn't have rights to read some tenant, in that case log warning but continue refresh
+      _log.warn "Forbidden to read the tenant, Message=#{err.message}"
```

Comment:
This warning needs to contain the tenant name as well as the keystone hostname so that users can find out what failed to connect.  It would also be helpful to show the collection type.